lib/api/controllers/documentController.js

```diff
@@ -419,26 +411,29 @@ class DocumentController extends BaseController {
    * @returns {Promise<Object>}
    */
   mDelete (request) {
+    let _errors;
+
     const
       deletedIds = [],
-      ids = this.getArrayParam(request, 'body.ids'),
-      refresh = this.getStringParam(request, 'args.refresh', 'false'),
+      ids = this.getBodyArray(request, 'ids'),
+      refresh = this.getString(request, 'refresh', 'false'),
       { index, collection } = this.getIndexAndCollection(request);
 
     return this.publicStorage.mDelete(index, collection, ids, { refresh })
       .then(({ documents, errors }) => {
+        _errors = errors;
+
         for (let i = 0; i < documents.length; i++) {
           deletedIds.push(documents[i]._id);
         }
 
-        if (errors.length > 0) {
-          request.setError(errorsManager.get('deletion_failed', errors));
-        }
-
         // @todo should be done in clientAdapter ?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)